java-squid/src/main/java/org/sonar/java/resolve/BytecodeVisitor.java

```diff
@@ -370,6 +399,39 @@ public void visitEnd() {
     }
   }
 
+
+
+  private static class TypeParameterDeclaration extends SignatureVisitor {
+
+    private final JavaSymbol symbol;
+
+    public TypeParameterDeclaration(JavaSymbol symbol) {
+      super(Opcodes.ASM5);
+      this.symbol = symbol;
+      if(symbol.isTypeSymbol()) {
+        ((JavaSymbol.TypeJavaSymbol) symbol).typeParameters = new Scope(symbol);
+      }else if (symbol.isMethodSymbol()) {
+        ((JavaSymbol.MethodJavaSymbol) symbol).typeParameters = new Scope(symbol);
+      }
+    }
+
+    @Override
+    public void visitFormalTypeParameter(String name) {
+      JavaSymbol.TypeVariableJavaSymbol typeVariableSymbol = new JavaSymbol.TypeVariableJavaSymbol(name, symbol);
+      if(symbol.isTypeSymbol()) {
+        JavaSymbol.TypeJavaSymbol classSymbol = (JavaSymbol.TypeJavaSymbol) symbol;
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Rename "classSymbol" which hides the field declared at line 43.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)